src/common/src/error.rs

```diff
@@ -195,12 +192,41 @@ pub struct RwError {
 
 impl From<RwError> for tonic::Status {
     fn from(err: RwError) -> Self {
-        match &*err.inner {
-            ErrorCode::ExprError(e) => tonic::Status::invalid_argument(e.to_string()),
-            ErrorCode::PermissionDenied(e) => tonic::Status::permission_denied(e),
-            ErrorCode::InternalError(e) => tonic::Status::internal(e),
-            _ => tonic::Status::internal(err.to_string()),
+        use tonic::Code;
+
+        let code = match &*err.inner {
+            ErrorCode::ExprError(_) => Code::InvalidArgument,
```

Comment:
ExprError => InvalidArgument doesn't look very correct 🤣

BTW, I was always wondering whether we should map application error code to tonic status code.. This is like the debate whether to use HTTP status code or simply use it as a wrapper and do everything in the payload. I don't have an answer. Just want to mention it. Let's just keep it until somebody is unhappy.

Reply:
I agree. I just leave the behavior unchanged in this PR. 😕

> whether we should map application error code to tonic status code.

+1. And the only place we refer to it is the error message of `Status`. 😄